src/GraphQL/Execution/ExecutionOptions.cs

```diff
@@ -62,5 +62,11 @@ public class ExecutionOptions
         /// Provides the ability to filter the schema upon introspection to hide types.
         /// </summary>
         public ISchemaFilter SchemaFilter { get; set; } = new DefaultSchemaFilter();
+
+        /// <summary>
+        /// Sets the service scope to be used for this request.  If not set, execution will default
```

Comment:
> If not set..

Do you assume that this is normal when this property is not set? And what should the client code in the resolver do in this case?

Reply:
For maximum compatibility, I designed it such that if the developer left it null, then all the ResolveFieldContext.Services fields would be null.  If the developer wished to used ResolveFieldContext.Services, they would set ExecutionOptions.Services.  It could grab the service provider from the schema, but that is a root service provider, and would not be scoped, so that is not a good option.  It could grab the service provider from the httpcontext, but that would rely on or assume that Asp.Net Core was in use

Reply:
You go through the same series of questions that I went through. Too many assumptions, too much manual work.